framework/base/Security.php

```diff
@@ -521,14 +521,16 @@ public function generateRandomKey($length = 32)
             if ($urandomFile !== null) {
                 // $length could be large so read using a loop.
                 $key = '';
+                $bytesLeft = $length;
                 do {
-                    $buffer = fread($urandomFile, $length);
+                    $buffer = fread($urandomFile, min($bytesLeft, 8092));
```

Comment:
@cebe Yes, you're right. Perhaps this is why I had the nagging feeling that I had forgotten something.